adapter-base/src/main/java/org/eclipse/hono/adapter/exception/DeviceDisabledOrNotRegisteredException.java

```diff
@@ -0,0 +1,37 @@
+/**
+ * Copyright (c) 2021 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.hono.adapter.exception;
+
+import org.eclipse.hono.client.ClientErrorException;
+
+/**
+ * An exception indicating that the device is either not registered or disabled.
+ */
+public class DeviceDisabledOrNotRegisteredException extends ClientErrorException {
```

Comment:
There is already a `RegistrationAssertionException` class in the parent package with a similar purpose.

Reply:
`RegistrationAssertionException` extends `AuthorizationException` and it is used in the AMQP and MQTT adapters while a device makes a connection request (`AbstractProtocolAdapterBase.checkDeviceRegistration`). As the parent class indicates, the error code is always `HTTP_UNAUTHORIZED`. 

In our case before processing messages, `AbstractProtocolAdapterBase.getRegistrationAssertion` is being invoked and the expected error code are `HTTP_NOT_FOUND` or `HTTP_FORBIDDEN`. Also I wanted to differentiate the device from the gateway. Hence I have created classes `DeviceDisabledOrNotRegisteredException` and `GatewayDisabledOrNotRegisteredException`

As I responded in the other comment, I would like to move all the exception classes under adapter module to the _exception_ package and if it is feasible and makes sense, I would replace the `RegistrationAssertionException` with the new ones in similar to the `AdapterDisabledException`.

Reply:
Ok, so `RegistrationAssertionException` should be replaced/adapted afterwards then.